specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-03-15-preview/inference.json

```diff
@@ -0,0 +1,816 @@
+                  },
+                  "required": [
+                    "object",
+                    "model",
+                    "data",
+                    "usage"
+                  ]
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/deployments/{deployment-id}/chat/completions": {
+      "post": {
+        "summary": "Creates a completion for the chat message",
+        "operationId": "ChatCompletions_Create",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "deployment-id",
+            "required": true,
+            "schema": {
+              "type": "string",
+              "description": "Deployment id of the model which was deployed."
+            }
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "required": true,
+            "schema": {
+              "type": "string",
+              "example": "2023-03-15-preview",
+              "description": "api version"
+            }
+          }
+        ],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "object",
+                "properties": {
+                  "messages": {
+                    "description": "The messages to generate chat completions for, in the chat format.",
+                    "type": "array",
+                    "minItems": 1,
+                    "items": {
+                      "type": "object",
+                      "properties": {
+                        "role": {
+                          "type": "string",
+                          "enum": [
+                            "system",
+                            "user",
+                            "assistant"
+                          ],
+                          "description": "The role of the author of this message."
+                        },
+                        "content": {
+                          "type": "string",
+                          "description": "The contents of the message"
+                        },
+                        "name": {
+                          "type": "string",
+                          "description": "The name of the user in a multi-user chat"
+                        }
+                      },
+                      "required": [
+                        "role",
+                        "content"
+                      ]
+                    }
+                  },
+                  "temperature": {
+                    "description": "What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.\nWe generally recommend altering this or `top_p` but not both.",
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 2,
```

Comment:
When I worked on this, I was wondering about Completions. We don't specify min/max there. Is this intended?

Reply:
The swagger usually gets copied unmodified from OpenAI, so the minimum was not specified for completions before. I guess it was added for the new chat completions and is just inconsistent with the other parts. I recommend to stick as close as possible to the OpenAI swagger and in case of change go through the breaking change process.